DependencyInjection/DoctrineExtension.php

```diff
@@ -544,8 +545,17 @@ protected function loadOrmEntityManager(array $entityManager, ContainerBuilder $
         $ormConfigDef = $container->setDefinition(sprintf('doctrine.orm.%s_configuration', $entityManager['name']), new ChildDefinition('doctrine.orm.configuration'));
         $ormConfigDef->addTag(IdGeneratorPass::CONFIGURATION_TAG);
 
+        $wrapLegacyCache     = true;
+        $metadataCacheSetter = 'setMetadataCacheImpl';
+
+        $metadataCacheType = $entityManager['metadata_cache']['type'] ?? 'pool';
+        if (method_exists(OrmConfiguration::class, 'setMetadataCache') && $metadataCacheType !== 'service') {
```

Comment:
@nicolas-grekas I believe this check is kosher. While users could theoretically change the class being used, ORM requires the configuration class to be of the instance we're checking here. Please let me know if this is not safe to do this way. Thanks!